R/system.R

```diff
@@ -1573,8 +1573,13 @@ read_delim_file <- function(file, delim, quote = '"',
     # if it's local file simply call readr::read_delim
     # reading through file() is to be able to read files with path that includes multibyte chars.
     # without it, error is thrown from inside read_delim.
-    readr::read_delim(file(file), delim, quote = quote, escape_backslash = escape_backslash, escape_double = escape_double, col_names = col_names, col_types = col_types,
-                      locale = locale, na = na, quoted_na = quoted_na, comment = comment, trim_ws = trim_ws, skip = skip, n_max = n_max, guess_max = guess_max, progress = progress)
+    if(stringi::stri_enc_mark(file) == "ASCII"){
+      readr::read_delim(file, delim, quote = quote, escape_backslash = escape_backslash, escape_double = escape_double, col_names = col_names, col_types = col_types,
```

Comment:
The only difference here is the first argument. I don't think you need to repeat the whole function call here.